dapr-spring/dapr-spring-boot-autoconfigure/src/test/java/io/dapr/spring/boot/autoconfigure/client/DaprClientAutoConfigurationTests.java

```diff
@@ -39,4 +44,26 @@ void daprClient() {
     contextRunner.run(context -> assertThat(context).hasSingleBean(DaprClient.class));
   }
 
+  @Test
+  @DisplayName("Should override properties when generating DaprClientBuilder")
+  void shouldOverridePropertiesWhenGeneratingDaprClientBuilder() {
+    PropertiesDaprConnectionDetails details = new PropertiesDaprConnectionDetails(
+        new DaprClientProperties(
+            "http://localhost", "localhost", 3500, 50001
+        )
+    );
+    contextRunner.withBean(DaprConnectionDetails.class, () -> details).run(context -> {
+
```

Comment:
@mcruzdev we might need to rethink this, relying on reflection is dangerous and I don't want to encourage such an approach.